app/models/user.rb

```diff
@@ -476,12 +482,31 @@ def active_flags(post)
     post.flags.where(user: self, status: nil)
   end
 
-  def do_soft_delete(attribute_to)
+  # Anonymizes the user (f.e., for the purpose of soft deletion)
+  # @param dirty [Boolean] if set to +false+, will persist the changes
+  def anonymize(dirty: false)
```

Comment:
suggestion: rename this parameter - my instinct for a `dirty` param is that it indicates a "dirty" operation, i.e. one with side-effects - meaning `true` would persist changes. Sounds like there are multiple interpretations of this - let's rename it something more descriptive. `persist_changes`, perhaps?